exercises/linked-list/canonical-data.json

```diff
@@ -0,0 +1,543 @@
+            "value": 41
+          },
+          {
+            "operation": "push",
+            "value": 59
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "push",
+            "value": 47
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop",
+            "value": 47
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "4e1948b4-514e-424b-a3cf-a1ebbfa2d1ad",
+      "description": "shifting to empty doesn't break the list",
+      "comments": [
+        "When not implemented correctly, head/tail reference will break when  ",
+        "the last item of the list is removed. This test adds an item after a ",
+        "list with at least two elements (different head than tail) has been  ",
+        "emptied out. If done correctly, the list will still work."
+      ],
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 41
+          },
+          {
+            "operation": "push",
+            "value": 59
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "push",
+            "value": 47
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop",
+            "value": 47
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "e8f7c600-d597-4f79-949d-8ad8bae895a6",
+      "description": "deletes the only element",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 61
+          },
+          {
+            "operation": "delete",
+            "value": 61
+          },
+          {
+            "operation": "count",
+            "expected": 0
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "fd65e422-51f3-45c0-9fd0-c33da638f89b",
+      "description": "deletes the element with the specified value from the list",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 71
+          },
+          {
+            "operation": "push",
+            "value": 83
+          },
+          {
+            "operation": "push",
+            "value": 79
+          },
+          {
+            "operation": "delete",
+            "value": 83
+          },
+          {
+            "operation": "count",
+            "expected": 2
+          },
+          {
+            "operation": "pop",
+            "expected": 79
+          },
+          {
+            "operation": "shift",
```

Comment:
although I think having both of these be `pop` or both `shift` would catch more mistakes (forgetting to link the elements on the two sides of the deleted element together), it is only a guess and I have no ground to demand changes based on just a guess.

Reply:
I don't know either! I can change it 🤷🏽 😄 .

We can also have _both_ test cases. What do you think?

Reply:
I'm the sort of person who likes to make my tests try to catch as many bugs as possible. So now that you've mentioned having both... I think I like that! Now I'm only guessing at what the possible bugs could be that a student might write, and I wouldn't know for sure since I'm not a mentor. But this is my best guess, and my suggestion based on that guess.

Reply:
I think your guess is good here. I'll add the extra test case.

I did try to add as many tests as possible that catch edge cases I've seen in the wild, and the one you've mentioned is one I didn't see/think of, but it's good to check!